src/pages/people-page/PeoplePage.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../../components/Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { useParams } from 'react-router-dom';
+
+export const PeoplePage = () => {
+  const [peopleList, setPeopleList] = useState<Person[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { peopleSlug } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(res => {
+        const peopleRes: Person[] = [];
+
+        res.map(person => {
+          const mother = res.find(p => p.name === person.motherName);
+          const father = res.find(p => p.name === person.fatherName);
+
+          peopleRes.push({ ...person, mother: mother, father: father });
+        });
+        setPeopleList(peopleRes);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        <div className="box table-container">
+          {isLoading && <Loader />}
+
+          {hasError && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {peopleList.length === 0 && !isLoading && (
+            <p data-cy="noPeopleMessage">There are no people on the server</p>
+          )}
+
+          {peopleList.length > 0 && (
```

Comment:
All there _if_ statements could be changed to a chained ternary to prevent case overlapping:

```
{isLoading ? (
    <Loader
) : hasError ? (
   ...
) : peopleList.length === 0 ? (
   ...
) : (
   <table ...
)}
```